commandline/src/test/java/gov/cms/qpp/conversion/CommandLineRunnerTest.java

```diff
@@ -0,0 +1,124 @@
+		runner.run();
+		Truth.assertThat(getLogs()).doesNotContain("A given template scope was invalid");
+	}
+
+	@Test
+	void testRunWithMissingFile() {
+		CommandLineRunner runner = new CommandLineRunner(line(INVALID_FILE));
+		runner.run();
+		Truth.assertThat(getLogs()).contains("Invalid or missing paths: [FILE]".replace("FILE", INVALID_FILE));
+	}
+
+	@JimfsTest
+	void testRunWithValidFile(FileSystem fileSystem) throws IOException {
+		CommandLineRunner runner = new CommandLineRunner(line("src/test/resources/valid-QRDA-III-abridged.xml"), fileSystem);
+		runner.run();
+		Truth.assertThat(Files.exists(fileSystem.getPath("valid-QRDA-III-abridged.qpp.json"))).isTrue();
+	}
+
+	@JimfsTest
+	void testRunWithValidFileSkipDefaults(FileSystem fileSystem) throws IOException {
+		CommandLineRunner runner = new CommandLineRunner(line("src/test/resources/valid-QRDA-III-abridged.xml",
+				"-" + CommandLineMain.SKIP_DEFAULTS), fileSystem);
+		runner.run();
+		Truth.assertThat(Files.exists(fileSystem.getPath("valid-QRDA-III-abridged.qpp.json"))).isTrue();
+	}
+
+	@JimfsTest
+	void testRunWithInvalidFile(FileSystem fileSystem) throws IOException {
+		CommandLineRunner runner = new CommandLineRunner(line("src/test/resources/qrda_bad_denominator.xml"), fileSystem);
+		runner.run();
+		Truth.assertThat(Files.exists(fileSystem.getPath("qrda_bad_denominator.err.json"))).isTrue();
+	}
+
+	@JimfsTest
+	void testRunWithInvalidFileWithoutValidation(FileSystem fileSystem) throws IOException {
+		CommandLineRunner runner = new CommandLineRunner(line("src/test/resources/qrda_bad_denominator.xml",
+				"-" + CommandLineMain.SKIP_VALIDATION), fileSystem);
+		runner.run();
+		Truth.assertThat(Files.exists(fileSystem.getPath("qrda_bad_denominator.qpp.json"))).isTrue();
+	}
+
+	@JimfsTest
+	void testRunWithValidFileGlobAtHeadInRoot(FileSystem fileSystem) throws IOException {
+		Files.copy(fileSystem.getPath("src/test/resources/valid-QRDA-III-abridged.xml"), fileSystem.getPath("valid-QRDA-III-abridged.xml"));
+		CommandLineRunner runner = new CommandLineRunner(line("*.xml"), fileSystem);
+		runner.run();
+		Truth.assertThat(Files.exists(fileSystem.getPath("valid-QRDA-III-abridged.qpp.json"))).isTrue();
+	}
+
+	@JimfsTest
+	void testRunWithValidFileGlobAtTailInRoot(FileSystem fileSystem) throws IOException {
+		Files.copy(fileSystem.getPath("src/test/resources/valid-QRDA-III-abridged.xml"), fileSystem.getPath("valid-QRDA-III-abridged.xml"));
+		CommandLineRunner runner = new CommandLineRunner(line("valid-QRDA-III-abridged.*"), fileSystem);
+		runner.run();
+		Truth.assertThat(Files.exists(fileSystem.getPath("valid-QRDA-III-abridged.qpp.json"))).isTrue();
+	}
+
+	@JimfsTest
+	void testRunWithValidGlobAllFiles(FileSystem fileSystem) throws IOException {
+		CommandLineRunner runner = new CommandLineRunner(line("src/test/resources/*"), fileSystem);
+		runner.run();
+		Truth.assertThat(Files.exists(fileSystem.getPath("valid-QRDA-III-abridged.qpp.json"))).isTrue();
```

Comment:
Can we assert that other files other than just `valid-QRDA-III-abridged` was converted?

Reply:
done